benches/sha256_hash.rs

```diff
@@ -0,0 +1,42 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use clvmr::sha2::Sha256;
+
+const BYTE_LENGTHS: [u8; 6] = [8, 16, 32, 64, 96, 128];
+const MAX_VAL: u8 = 250;
+
+fn gen_bytes(value: u8, amount: u8) -> Vec<u8> {
+    let mut bytes = Vec::new();
+    for _ in 0..amount {
+        bytes.push(value);
+    }
+    bytes
+}
+
+fn hash_bytes(bytes: &[u8]) -> [u8; 32] {
+    let mut sha256 = Sha256::new();
+    sha256.update(bytes);
+    sha256.finalize()
+}
+
+fn sha256_hash_benchmark(c: &mut Criterion) {
+    // setup benchmark
+    let mut group = c.benchmark_group("sha256_hash");
+
+    group.bench_function("hash_benchmark", |b| {
+        b.iter(|| {
+            // this figures out how many iterations to run.
+            for val in 0..MAX_VAL {
+                for len in BYTE_LENGTHS {
+                    let bytes = gen_bytes(val, len);
+                    hash_bytes(&bytes);
```

Comment:
there's a chance you need to use `black_box()` here, to prevent the optimizer removing all computation here

Reply:
https://docs.rs/criterion/latest/criterion/fn.black_box.html#:~:text=Function%20criterion%3A%3Ablack_box&text=A%20function%20that%20is%20opaque,prevent%20code%20from%20being%20eliminated.